authors/apps/articles/tests/test_rate_article.py

```diff
@@ -0,0 +1,118 @@
+from django.urls import reverse
+from rest_framework import status
+
+from .base_setup import Base
+
+from authors.apps.authentication.backends import generate_jwt_token
+from authors.apps.articles.models import ArticleRating
+
+
+class ArticleRatingTests(Base):
+    def setUp(self):
+        """ Set up test environment"""
+        super().setUp()
+        self.res = self.client.post(self.article_url, self.article_data,
+                                    format="json", **self.headers)
+        self.slug = self.res.data['slug']
+
+        self.violate_max_value_error_message = \
+            'Ensure this value is less than or equal to 5.'
+
+        self.violate_min_value_error_message = \
+            'Ensure this value is greater than or equal to 1.'
+
+        self.violate_empty_value_error_message = \
+            'Please fill in the rating'
+
+        self.non_existent_article_message = \
+            'That article does not exist'
+
+    def tearDown(self):
+        """ Tear dowm test environment"""
+        super().tearDown()
+
+    def test_successful_article_rate(self):
+        """
+        Tests that a user can rate a specific article
+        """
+        ratings_count = ArticleRating.objects.count()
+        response = self.client.post(
+            reverse('articles:rate', kwargs={'slug': self.slug}),
+            {'rating': 4},
+            format="json",
+            **self.headers)
+        self.assertEqual(response.status_code, status.HTTP_201_CREATED)
+        self.assertEqual(ArticleRating.objects.count(), ratings_count+1)
+
+    def test_successful_article_rate_update(self):
+        """
+        Tests that a user can rate a specific article
+        """
+
+        self.client.post(
+            reverse('articles:rate', kwargs={'slug': self.slug}),
+            {'rating': 4},
+            format="json",
+            **self.headers)
+
+        ratings_count = ArticleRating.objects.count()
+        update_response = self.client.post(
+            reverse('articles:rate', kwargs={'slug': self.slug}),
+            {'rating': 5},
+            format="json",
+            **self.headers)
+        self.assertEqual(update_response.status_code, status.HTTP_201_CREATED)
+        self.assertEqual(ArticleRating.objects.count(), ratings_count)
+
+    def test_unsuccessful_rating_with_negative_rate_value(self):
+        """
+        Tests that a user can rate a specific article
+        """
+        response = self.client.post(
+            reverse('articles:rate', kwargs={'slug': self.slug}),
+            {'rating': -4},
+            format="json",
+            **self.headers)
+        self.assertEqual(response.status_code, status.HTTP_400_BAD_REQUEST)
+        self.assertEqual(str(response.data['errors']['rating'][0]),
+                         self.violate_min_value_error_message)
+
+    def test_unsuccessful_rating_with_rate_value_more_than_five(self):
+        """
+        Tests that a user can rate a specific article
```

Comment:
Remember to update the doc strings for your tests; They seem to be similar even when you have indicated to be testing both positive and negative outcomes.

Reply:
This has been corrected